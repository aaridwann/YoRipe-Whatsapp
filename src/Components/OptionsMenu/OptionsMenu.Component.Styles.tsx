import { StyleSheet } from 'react-native';

import { scaleSize } from '../../Utils/Size/size.utils';
import Constants from '../../Constants/Constants';

export const style = StyleSheet.create({
    container: {
        padding: scaleSize(8),
        justifyContent: 'center',
        alignContent: 'center',
        backgroundColor: Constants.Colors.GREEN,
        shadowColor: '#000000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.3,
        shadowRadius: 2,
        elevation: 4,
        borderRadius: 5,
    },
    listMenu: {
        margin: scaleSize(8),
        fontWeight: '600',
        fontSize: scaleSize(13),
        color: Constants.Colors.WHITE,
    },
});
