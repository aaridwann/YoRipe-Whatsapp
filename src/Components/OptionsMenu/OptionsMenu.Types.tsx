export interface Props {
    isShow: boolean;
    list: Array<string>;
    select: (name: string) => void;
}
