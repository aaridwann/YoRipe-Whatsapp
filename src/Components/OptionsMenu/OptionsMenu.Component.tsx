import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Text } from 'react-native';
import * as Animatable from 'react-native-animatable';

import { style } from './OptionsMenu.Component.Styles';
import { Props } from './OptionsMenu.Types';

const _renderListMenu = (props: Props) =>
    props.list.map((menu) => (
        <React.Fragment key={menu}>
            <TouchableOpacity onPress={() => props.select(menu)}>
                <Text style={style.listMenu}>{menu}</Text>
            </TouchableOpacity>
        </React.Fragment>
    ));

const OptionsMenu = (props: Props) => (
    <Animatable.View
        useNativeDriver={true}
        duration={500}
        animation={props.isShow ? 'fadeInRightBig' : 'fadeOutRightBig'}
        style={style.container}
    >
        {_renderListMenu(props)}
    </Animatable.View>
);

export default OptionsMenu;

OptionsMenu.defaultProps = {
    isShow: false,
};
