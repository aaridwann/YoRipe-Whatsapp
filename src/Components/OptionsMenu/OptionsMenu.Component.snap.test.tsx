import { render } from '@testing-library/react-native';

import OptionsMenu from './OptionsMenu.Component';

describe('Options Menu', () => {
    it('Should render with correct props', () => {
        const Props = {
            isShow: true,
            list: ['setting', 'profile'],
            select: jest.fn(),
        };

        expect(render(OptionsMenu(Props))).toMatchSnapshot();
    });
});
