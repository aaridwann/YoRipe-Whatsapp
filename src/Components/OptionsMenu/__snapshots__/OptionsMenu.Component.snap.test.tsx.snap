// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Menu Should render with correct props 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#075e55",
      "borderRadius": 5,
      "elevation": 4,
      "justifyContent": "center",
      "opacity": 0,
      "padding": 16,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "transform": [
        {
          "translateX": 500,
        },
      ],
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 26,
          "fontWeight": "600",
          "margin": 16,
        }
      }
    >
      setting
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 26,
          "fontWeight": "600",
          "margin": 16,
        }
      }
    >
      profile
    </Text>
  </View>
</View>
`;
